equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/model/Crm4Model.java

```diff
@@ -18,4 +21,23 @@ public class Crm4Model implements CrmFormModelInterface {
 
     @JsonProperty("linkedAttachments")
     private CrmEvidenceFilesModel evidenceFiles;
+
+    public void addFileKeyToFurtherInfo(List<CrmFurtherInfoAttachmentsModel> furtherInfoAttachments) {
+        for(CrmFurtherInfoAttachmentsModel furtherInfoAttachment: furtherInfoAttachments){
+            if(StringUtils.isNotEmpty(furtherInfoAttachment.getRetrieve()) && StringUtils.isNotEmpty(furtherInfoAttachment.getName())){
+                furtherInfoAttachment.setFileKey(retrieveFileKey(furtherInfoAttachment.getName()));
+            }
+        }
+    }
+    public String retrieveFileKey(String name) {
```

Comment:
suggestion - possible to rename to `filename` for readability

Reply:
Done